diff --git a/src/Backends/WaylandBackend.cpp b/src/Backends/WaylandBackend.cpp
index 664ed0eac..57207a869 100644
--- a/src/Backends/WaylandBackend.cpp
+++ b/src/Backends/WaylandBackend.cpp
@@ -678,6 +678,7 @@ namespace gamescope
         wp_fractional_scale_manager_v1 *GetFractionalScaleManager() const { return m_pFractionalScaleManager; }
         xdg_toplevel_icon_manager_v1 *GetToplevelIconManager() const { return m_pToplevelIconManager; }
         libdecor *GetLibDecor() const { return m_pLibDecor; }
+        wp_image_description_v1 *CreateManualWindowsSCRGB();
 
         void UpdateFullscreenState();
 
@@ -1428,10 +1429,25 @@ namespace gamescope
 
                     if ( oState->eColorspace == GAMESCOPE_APP_TEXTURE_COLORSPACE_SCRGB )
                     {
-                        m_pCurrentImageDescription = wp_color_manager_v1_create_windows_scrgb( m_pBackend->GetWPColorManager() );
+                        if ( Algorithm::Contains( m_pBackend->m_WPColorManagerFeatures.eFeatures, WP_COLOR_MANAGER_V1_FEATURE_WINDOWS_SCRGB ))
+                        {
+                            m_pCurrentImageDescription = wp_color_manager_v1_create_windows_scrgb( m_pBackend->GetWPColorManager() );
+                        }
+                        else
+                        {
+                            xdg_log.warnf( "Attempting to transition to Windows scRGB, but it is not supported by the compositor." );
+                            m_pCurrentImageDescription = m_pBackend->CreateManualWindowsSCRGB();
+                        }
+
                     }
                     else if ( oState->eColorspace == GAMESCOPE_APP_TEXTURE_COLORSPACE_HDR10_PQ )
                     {
+                        if ( !( Algorithm::Contains( m_pBackend->m_WPColorManagerFeatures.ePrimaries, WP_COLOR_MANAGER_V1_PRIMARIES_BT2020 ) &&
+                                Algorithm::Contains( m_pBackend->m_WPColorManagerFeatures.eTransferFunctions, WP_COLOR_MANAGER_V1_TRANSFER_FUNCTION_ST2084_PQ ) ) )
+                        {
+                            xdg_log.errorf( "Attempting to transition to HDR10 PQ, but it is not supported by the compositor." );
+                        }
+
                         wp_image_description_creator_params_v1 *pParams = wp_color_manager_v1_create_parametric_creator( m_pBackend->GetWPColorManager() );
 
                         double flScale = cv_wayland_hdr10_saturation_scale;
@@ -2006,6 +2022,37 @@ namespace gamescope
 
                 return true;
             }();
+
+            // HDR10.
+            if ( Algorithm::Contains( m_WPColorManagerFeatures.ePrimaries, WP_COLOR_MANAGER_V1_PRIMARIES_BT2020 ) &&
+                    Algorithm::Contains( m_WPColorManagerFeatures.eTransferFunctions, WP_COLOR_MANAGER_V1_TRANSFER_FUNCTION_ST2084_PQ ) )
+            {
+                wp_image_description_creator_params_v1 *pParams = wp_color_manager_v1_create_parametric_creator( m_pWPColorManager );
+                wp_image_description_creator_params_v1_set_primaries_named( pParams, WP_COLOR_MANAGER_V1_PRIMARIES_BT2020 );
+                wp_image_description_creator_params_v1_set_tf_named( pParams, WP_COLOR_MANAGER_V1_TRANSFER_FUNCTION_ST2084_PQ );
+                m_pWPImageDescriptions[ GAMESCOPE_APP_TEXTURE_COLORSPACE_HDR10_PQ ] = wp_image_description_creator_params_v1_create( pParams );
+            }
+            else
+            {
+                xdg_log.warnf( "Compositor does not support HDR10 PQ." );
+            }
+
+            // scRGB
+            if ( Algorithm::Contains( m_WPColorManagerFeatures.eFeatures, WP_COLOR_MANAGER_V1_FEATURE_WINDOWS_SCRGB ) )
+            {
+                m_pWPImageDescriptions[ GAMESCOPE_APP_TEXTURE_COLORSPACE_SCRGB ] = wp_color_manager_v1_create_windows_scrgb( m_pWPColorManager );
+            }
+            // scRGB-ish
+            else if ( Algorithm::Contains( m_WPColorManagerFeatures.eTransferFunctions, WP_COLOR_MANAGER_V1_TRANSFER_FUNCTION_EXT_LINEAR ) )
+            {
+                xdg_log.warnf( "Compositor does not support Windows scRGB. Attempting to create parametrically." );
+                m_pWPImageDescriptions[ GAMESCOPE_APP_TEXTURE_COLORSPACE_SCRGB ] = CreateManualWindowsSCRGB();
+            }
+            else
+            {
+                xdg_log.warnf( "Compositor does not support Windows scRGB." );
+            }
+
         }
 
         m_pLibDecor = libdecor_new( m_pDisplay, &s_LibDecorInterface );
@@ -2409,6 +2456,18 @@ namespace gamescope
         }
     }
 
+    wp_image_description_v1 *CWaylandBackend::CreateManualWindowsSCRGB()
+    {
+        // Attempt to manually create something approximating an scRGB image description:
+        // https://invent.kde.org/plasma/kwin/-/blob/master/src/wayland/colormanagement_v1.cpp#L94-107
+        // https://wayland.app/protocols/color-management-v1#wp_color_manager_v1:request:create_windows_scrgb
+        wp_image_description_creator_params_v1 *pParams = wp_color_manager_v1_create_parametric_creator( m_pWPColorManager );
+        wp_image_description_creator_params_v1_set_primaries_named( pParams, WP_COLOR_MANAGER_V1_PRIMARIES_SRGB );
+        wp_image_description_creator_params_v1_set_tf_named( pParams, WP_COLOR_MANAGER_V1_TRANSFER_FUNCTION_EXT_LINEAR );
+        wp_image_description_creator_params_v1_set_luminances( pParams, 0, 10000, 203 );
+        return wp_image_description_creator_params_v1_create( pParams );
+    }
+
     /////////////////////
     // Wayland Callbacks
     /////////////////////
@@ -2476,19 +2535,6 @@ namespace gamescope
         {
             m_pWPColorManager = (wp_color_manager_v1 *)wl_registry_bind( pRegistry, uName, &wp_color_manager_v1_interface, 1u );
             wp_color_manager_v1_add_listener( m_pWPColorManager, &s_WPColorManagerListener, this );
-
-            // HDR10.
-            {
-                wp_image_description_creator_params_v1 *pParams = wp_color_manager_v1_create_parametric_creator( m_pWPColorManager );
-                wp_image_description_creator_params_v1_set_primaries_named( pParams, WP_COLOR_MANAGER_V1_PRIMARIES_BT2020 );
-                wp_image_description_creator_params_v1_set_tf_named( pParams, WP_COLOR_MANAGER_V1_TRANSFER_FUNCTION_ST2084_PQ );
-                m_pWPImageDescriptions[ GAMESCOPE_APP_TEXTURE_COLORSPACE_HDR10_PQ ] = wp_image_description_creator_params_v1_create( pParams );
-            }
-
-            // scRGB
-            {
-                m_pWPImageDescriptions[ GAMESCOPE_APP_TEXTURE_COLORSPACE_SCRGB ] = wp_color_manager_v1_create_windows_scrgb( m_pWPColorManager );
-            }
         }
         else if ( !strcmp( pInterface, zwp_pointer_constraints_v1_interface.name ) )
         {
